src/openms/source/FORMAT/CachedMzML.cpp

```diff
@@ -83,7 +84,7 @@ namespace OpenMS
     ifs_.open(filename_cached_.c_str(), std::ios::binary);
 
     // load the meta data from disk
-    MzMLFile().load(filename, meta_ms_experiment_);
```

Comment:
here it must 100% be mzML

Reply:
Must it? We use MzML in the name for this file type, but it doesn't actually need to be an MzML, its really a cached `MSExperiment`. The only thing we are loading with this call is the meta data.

Reply:
I think the meta data is only fully supported in mzML. Not sure what happens if a mgf is passed …

Reply:
Hmm, good point. It's certainly not high on my list to check something like mgf. It does make me wonder if renaming the class would make some sense in the future.

Reply:
yeah naming is a bit an issue in OpenMS. Also having the FileHandler (which is kind of a factory that creates and executes concrete reader) and XMLFileHandler (which is the part that manages the XML related parsing etc.) sound very similar but are conceptually totally different.